File: py/0Intro.py
# CLI usage: 
# node 0Intro.js x-y-z/a-b-c
#
# File usage:
# x y z 
# a b c

import sys, importlib.util

specFile = importlib.util.spec_from_file_location('file', '/home/ubuntu/Dev/algs-ds/tools/file.py')
file = importlib.util.module_from_spec(specFile)
specFile.loader.exec_module(file)

specCli = importlib.util.spec_from_file_location('cli', '/home/ubuntu/Dev/algs-ds/tools/cli.py')
cli = importlib.util.module_from_spec(specCli)
specCli.loader.exec_module(cli)

if len(sys.argv) > 1:
    cli.readArgTo2DArray(sys.argv[1])
else:
    file.readLinesTo2DArray('asdfasdfasdf')

File: js/0Intro.js
// CLI usage: 
// node 0Intro.js x-y-z/a-b-c
//
// File usage:
// x y z 
// a b c

var file = require('/home/ubuntu/Dev/algs-ds/tools/file.js');
var cli = require('/home/ubuntu/Dev/algs-ds/tools/cli.js');

const INPUT_FILE_PATH = '/home/ubuntu/Dev/algs-ds/input/0intro3.txt'

if (process.argv.length > 2) {
   cli.readArgTo2DArray(process.argv[2], function (cliInput) {
        intro(cliInput);
   });

} else {
    file.readLinesTo2DArray(INPUT_FILE_PATH, function (fileInput) {
        intro(fileInput);
    });
}

function intro (inputs) {
    console.log('A.) Full 2D Array');
    console.log(inputs);
    console.log('B.) Each Row');
    inputs.map(i => {
        console.log(i);
    });
    console.log('C.) Each Cell');
    inputs.map(i => {
        i.map(j => {
            console.log(j);
        });
    })
}